ext/node/ops.rs

```diff
@@ -470,7 +470,13 @@ where
   {
     modules_path
   } else {
-    path_resolve(vec![modules_path, name])
+    let orignal = modules_path.clone();
+    let mod_dir = path_resolve(vec![modules_path, name]);
+    if Path::new(&mod_dir).is_dir() {
+      mod_dir
+    } else {
+      orignal
+    }
```

Comment:
This is the fix here. I'm not too happy about it because it adds a file system check, but it does solve the issue.

The reason I'm not happy about it is that the resolution code expects node_module paths everywhere like `/my-project/node_modules`. But in this [branch here](https://github.com/denoland/deno/blob/c162647020c26de1d27064b488ca11525f1bc4bf/ext/node/polyfills/01_require.js#L618-L626) we ignore that assumption and push a module directory to the list of paths instead (`/my-project/node_modules/my-module`). Then we reach this section here where we'd incorrectly append the package name to the path again resulting in `/my-project/node_modules/my-module/my-module` which doesn't exist and therefore panics.

Reply:
Good catch. IIRC this was done to support CJS in situation where we use "global cache" (ie. not creating local `node_modules` directory), but appears that it's wrong. I'm happy to completely reorganize this bit of code so we don't have to do `is_dir()` check.